dev/checkstyle.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+#
+# Copyright (c) 2024, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+#!/bin/bash
+
+set -ex
+
+# Assuming you are in the root of your git repository
+MODIFIED_FILES=$(git diff --name-only "origin/${GITHUB_BASE_REF}")
```

Comment:
agree with Jason, We would like to not only check style for incremental changes.

and this script is current for github action only with the ENV provided by github. 
can we just check against all eligible code in the change?

and we do not need below in action def then
```
      with:
        submodules: true
        fetch-depth: 0

    - name: Fetch branches and tags
      run: git fetch --all
```